nautobot_golden_config/nornir_plays/config_intended.py

```diff
@@ -31,14 +32,19 @@
 
 
 def run_template(  # pylint: disable=too-many-arguments
-    task: Task, logger, global_settings, job_result, jinja_root_path, intended_root_folder
+    task: Task, logger, global_settings, nautobot_job, jinja_root_path, intended_root_folder
 ) -> Result:
     """Render Jinja Template.
 
     Only one template is supported, so the expectation is that that template includes all other templates.
 
     Args:
         task (Task): Nornir task individual object
+        logger (NornirLogger): Logger to log messages to.
+        global_settings (GoldenConfigSetting): The settings for GoldenConfigPlugin.
+        nautobot_job (Result): The Nautobot Job instance being ran.
```

Comment:
`job_result` is the term used in nautobot-core, any reason to not use this term?

Reply:
This is called here https://github.com/nautobot/nautobot-plugin-golden-config/blob/9b40247df2956f91e9478ff0347cac9cd230e70b/nautobot_golden_config/jobs.py#L139

The first argument is `self`, and self is an instance of `nautobot.extras.jobs.Job` https://github.com/nautobot/nautobot-plugin-golden-config/blob/9b40247df2956f91e9478ff0347cac9cd230e70b/nautobot_golden_config/jobs.py#L6 https://github.com/nautobot/nautobot-plugin-golden-config/blob/9b40247df2956f91e9478ff0347cac9cd230e70b/nautobot_golden_config/jobs.py#L106

JobResult is a different class found at `nautobot.extras.models.models.JobResult`, so the proposed name change is to hopefully communicate more effectively what the object is that is being passed to the function.


As for the variable name `job_result`, I did a search in the Nautobot codebase, and I only see that used with the JobResult class. For example: [job_result = JobResult.objects.get(pk=job_result_pk)](https://github.com/nautobot/nautobot/blob/1da6d03d0c30ab7f603695665972d758522aaaa4/nautobot/extras/datasources/git.py#L60)

The one exception was in the JobResult class itself, which is basically doing the same thing: [job_result = cls.objects.create(name=name, obj_type=obj_type, user=user, job_id=uuid.uuid4())](https://github.com/nautobot/nautobot/blob/b64919f88b83a4936cbf93fd9829f054363b8e2a/nautobot/extras/models/models.py)

I guess this brings up the question, should this expect a Job or a JobResult instance?

Reply:
I missed a step in this, the argument is passed from the `config_intended` function, which is where my explanation starts

Reply:
I thought I responded, but I guess not. I appreciate your detailed response to help me understand.

This may have downstream issues with nornir-nautobot, specifically with https://github.com/nautobot/nornir-nautobot/blob/a0f7ed722f96ce04f96de63ff8329ba883cd260f/nornir_nautobot/utils/logger.py#L11

Reply:
The logger sets this on instantiation, [self.job_result = job_result](https://github.com/nautobot/nornir-nautobot/blob/a0f7ed722f96ce04f96de63ff8329ba883cd260f/nornir_nautobot/utils/logger.py#L11).

The logger is instantiated in `config_intended` function below, and then passed in here. The `job_result` arg is passed in as a positional arg and not a keyword arg, so it seems to me that it should still work: https://github.com/nautobot/nautobot-plugin-golden-config/blob/a9e76ad92efd7e32cc9d68dd15f91cc2735d90b5/nautobot_golden_config/nornir_plays/config_intended.py#L109

Do you see something that I am missing?

Reply:
Action item, follow up with fix in nornir-nautobot.

Reply:
https://github.com/nautobot/nornir-nautobot/pull/46